File: application/credit-card-enrollment-backend-in-java/src/main/java/cloud/ambar/common/event/Event.java
package cloud.ambar.common.event;

import lombok.Getter;
import lombok.NonNull;
import lombok.experimental.SuperBuilder;

import java.time.Instant;

@SuperBuilder
@Getter
public abstract class Event {
    @NonNull protected String eventId;

    @NonNull protected Integer aggregateVersion;

    @NonNull protected String aggregateId;

    @NonNull protected String causationId;

    @NonNull protected String correlationId;

    @NonNull protected Instant recordedOn;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Common/Events/Event.cs
namespace CreditCardEnrollment.Common.Events;

public abstract class Event
{
    public string EventId { get; set; } = string.Empty;
    public string AggregateId { get; set; } = string.Empty;
    public int AggregateVersion { get; set; }
    public string CorrelationId { get; set; } = string.Empty;
    public string CausationId { get; set; } = string.Empty;
    public DateTime RecordedOn { get; set; }
}
